### PR TITLE
[cli] add `--deprecated` option to `vc project ls` command

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Adds a `--deprecated` option to the `vc project ls` command that will only show projects currently running on a soon-to-be-deprecated Node.js version.

It also adds additional output providing more information to the user about what versions and a link to our documentation so they can learn more.

Example:

Screenshot 2023-12-07 at 15 01 22

### CODE REVIEW

**Review by trek — APPROVED** [review]
Love it!

File: packages/cli/src/commands/project/list.ts [hunk 1]
```diff
@@ -48,10 +51,38 @@ export default async function list(
...
+  if (deprecated) {
+    const upcomingDeprecationVersions = NODE_VERSIONS.filter(
+      nodeVersion =>
+        nodeVersion.discontinueDate &&
+        nodeVersion.discontinueDate.valueOf() > Date.now()
+    );
+    const upcomingDeprecationVersionsList = upcomingDeprecationVersions.map(
+      nodeVersion => nodeVersion.range
+    );
+    console.log(projectList);
```
Comment (cb1kenobi) [inline packages/cli/src/commands/project/list.ts:63]: Remove debug `console.log()`.

File: packages/cli/src/commands/project/list.ts [hunk 2]
```diff
@@ -48,10 +51,38 @@ export default async function list(
...
+    projectList = projectList.filter(
+      project =>
+        project.nodeVersion &&
+        upcomingDeprecationVersionsList.includes(project.nodeVersion)
+    );
+
+    output.log(
+      `The following Node.js versions will be deprecated soon: ${upcomingDeprecationVersionsList.join(
```
Comment (cb1kenobi) [inline packages/cli/src/commands/project/list.ts:71]: I feel this needs to be called out better. Either an `!` or colored yellow or something.

File: packages/cli/test/unit/commands/project.test.ts [hunk 3]
```diff
@@ -49,6 +45,57 @@ describe('project', () => {
...
+      jest.useFakeTimers().setSystemTime(new Date('2023-12-08'));
+
+      const user = useUser();
+      useTeams('team_dummy');
+      const project = useProject({
+        ...defaultProject,
+        nodeVersion: '16.x',
+      });
+
+      console.log(project);
```
Comment (EndangeredMassa) [inline packages/cli/test/unit/commands/project.test.ts:58]: remove?

**Pull request merged**